pkg/providers/validator/validate.go

```diff
@@ -10,11 +10,43 @@ import (
 	"github.com/aws/eks-anywhere/pkg/providers"
 )
 
-// TODO: update vsphere, cloudstack, tinkerbell validators to use this common validation method.
-func ValidateControlPlaneIpUniqueness(cluster *v1alpha1.Cluster, netClient networkutils.NetClient) error {
+// DefaultIPValidator defines the struct for the default IPValidator.
+type DefaultIPValidator struct {
+	netClient networkutils.NetClient
+}
+
+// IPValidator defines an interface for the methods to validate the control plane IP.
+type IPValidator interface {
+	ValidateControlPlaneIPUniqueness(cluster *v1alpha1.Cluster, opts ...IPValidatorOpt) error
+}
+
+// IPValidatorOpt is the type for optional IPValidator configurations.
+type IPValidatorOpt func(e *DefaultIPValidator)
+
+// CustomNetClient passes in a custom net client to the IPValidator.
+func CustomNetClient(netClient networkutils.NetClient) IPValidatorOpt {
+	return func(d *DefaultIPValidator) {
+		d.netClient = netClient
+	}
+}
+
+// NewIPValidator initializes a new IPValidator object.
+func NewIPValidator() *DefaultIPValidator {
+	return &DefaultIPValidator{
+		netClient: &networkutils.DefaultNetClient{},
+	}
+}
+
+// ValidateControlPlaneIPUniqueness checks whether or not the control plane endpoint defined
+// in the cluster spec is available.
+func (v *DefaultIPValidator) ValidateControlPlaneIPUniqueness(cluster *v1alpha1.Cluster, opts ...IPValidatorOpt) error {
```

Comment:
I would suggest passing these options to the constructor, not to this method.
It keeps the interface cleaner, since it won't depend on types from the package. And it's also thread safe since you can't modify `DefaultIPValidator` once it has been constructed.